crates/primitives/src/l1.rs

```diff
@@ -352,6 +352,28 @@ impl BorshDeserialize for BitcoinAddress {
     }
 }
 
+impl PartialEq for BitcoinAddress {
+    fn eq(&self, other: &Self) -> bool {
+        self.address
+            .script_pubkey()
+            .eq(&other.address.script_pubkey())
+    }
+}
```

Comment:
This also needs to include the network field, missed it the previous time around.